### PR TITLE
pkg/api: Move bindings inside imports

### PR DESCRIPTION
**Pull request by orozery**
This PR moves bindings to be included in the import spec. The bindings API is removed.

### CODE REVIEW

File: pkg/controlplane/rest/import.go [hunk 1]
```diff
@@ -31,14 +31,22 @@ type importHandler struct {
...
 	return &v1alpha1.Import{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      imp.Name,
 			Namespace: namespace,
 		},
 		Spec: v1alpha1.ImportSpec{
-			Port:       imp.ImportSpec.Service.Port,
-			TargetPort: imp.Port,
+			Port:       imp.Port,
+			TargetPort: imp.TargetPort,
```
Comment (kfirtoledo) [inline pkg/controlplane/rest/import.go:48]: What happens if the targetPort is already used?

File: pkg/api/types.go [hunk 2]
```diff
@@ -87,34 +87,23 @@ type Import struct {
 	Name string
 	// Spec represents the attributes of the import service.
 	Spec ImportSpec
-	// Status field contains the import service status.
-	Status ImportStatus
```
Comment (kfirtoledo) [inline pkg/api/types.go:91]: Why did you remove the status?

File: pkg/api/types.go [hunk 3]
```diff
@@ -87,34 +87,23 @@ type Import struct {
...
 }
 
-// ImportSpec contains all the import service attributes.
-type ImportSpec struct {
```
Comment (kfirtoledo) [inline pkg/api/types.go:95]: Shouldn't be part of importspec?

File: pkg/controlplane/rest/import.go [hunk 4]
```diff
@@ -31,14 +31,22 @@ type importHandler struct {
 }
 
 func toK8SImport(imp *store.Import, namespace string) *v1alpha1.Import {
+	sources := make([]v1alpha1.ImportSource, len(imp.Peers))
+	for i, pr := range imp.Peers {
+		sources[i].Peer = pr
+		sources[i].ExportName = imp.Name
```
Comment (kfirtoledo) [inline pkg/controlplane/rest/import.go:37]: Do I need always to specify the ExportName and ExportNamespace? what happens if we don't?

File: tests/e2e/k8s/test_basic.go [hunk 5]
```diff
@@ -43,7 +43,7 @@ func (s *TestSuite) TestConnectivityCRD() {
 			Name: httpEchoService.Name,
 			Port: 80,
 		}
-		require.Nil(s.T(), cl[1].CreateImportCRD(importedService, cl[0], httpEchoService.Name))
+		require.Nil(s.T(), cl[1].CreateImport(importedService, cl[0], httpEchoService.Name))
```
Comment (kfirtoledo) [inline tests/e2e/k8s/test_basic.go:47]: You didn't want to combine the tests? with CRD or without?

Comment (orozery) [reply tests/e2e/k8s/test_basic.go:47]: I will do it in a separate PR that enhances tests.

File: pkg/controlplane/rest/import.go (hunk 4 above)
Comment (orozery) [reply pkg/controlplane/rest/import.go:37]: The `Import` API only includes an array of peers.
Here, we translate this array to `ImportSource`, which requires not just the peer, but also the export name and namespace.
In the old API (which is kept here), the export name must match the import name, and there's no namespaces.

File: pkg/api/types.go (hunk 2 above)
Comment (orozery) [reply pkg/api/types.go:91]: Since it is not really used.

**Review by kfirtoledo — APPROVED** [review]
